libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/task/RemoveTilesTask.kt

```diff
@@ -1,20 +1,35 @@
 package com.mapbox.navigation.route.onboard.task
 
-import android.os.AsyncTask
 import com.mapbox.geojson.Point
+import com.mapbox.navigation.navigator.MapboxNativeNavigator
 import com.mapbox.navigation.route.onboard.OnOfflineTilesRemovedCallback
-import com.mapbox.navigator.Navigator
+import com.mapbox.navigation.utils.thread.ThreadController
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 internal class RemoveTilesTask(
-    private val navigator: Navigator,
+    private val navigator: MapboxNativeNavigator,
     private val tilePath: String,
     private val southwest: Point,
     private val northeast: Point,
     private val callback: OnOfflineTilesRemovedCallback
-) : AsyncTask<Void, Void, Long>() {
+) {
 
-    override fun doInBackground(vararg paramsUnused: Void): Long =
-        navigator.removeTiles(tilePath, southwest, northeast)
+    private val jobControl = ThreadController.getMainScopeAndRootJob()
 
-    public override fun onPostExecute(numberOfTiles: Long) = callback.onRemoved(numberOfTiles)
+    fun launch() {
+        jobControl.scope.launch {
+            val numberOfTiles = withContext(Dispatchers.Default) {
```

Comment:
Should we use the dispatcher from `ThreadController` instead?

Reply:
done